modules/cherry-post-types/cherry-post-types.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * @package    Cherry_Framework
+ * @subpackage Class
+ * @author     Cherry Team <cherryframework@gmail.com>
+ * @copyright  Copyright (c) 2012 - 2016, Cherry Team
+ * @link       http://www.cherryframework.com/
+ * @license    http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ */
+
+// If this file is called directly, abort.
+if ( ! defined( 'WPINC' ) ) {
+	die;
+}
+
+/**
+ * Cherry post types class
+ * Example usage:
+ * $this->core->modules['cherry-post-types']->create(
+ *			'property',
+ *			'Property',
+ *			'Properties'
+ *		)->font_awesome_icon( 'f1ad' );
+ */
+class Cherry_Post_Types implements I_Module {
+	/**
+	 * Module version
+	 *
+	 * @var string
+	 */
+	public $module_version = '1.0.0';
+
+	/**
+	 * Module slug
+	 *
+	 * @var string
+	 */
+	public $module_slug = 'cherry-post-types';
+
+	/**
+	 * Default post type arguments
+	 *
+	 * @var null
+	 */
+	private $defaults = null;
+
+	/**
+	 * Created popst types list
+	 *
+	 * @var array
+	 */
+	public static $created_post_types = array();
+
+	/**
+	 * Module directory
+	 *
+	 * @since 1.0.0
+	 * @var string
+	 */
+	private $module_directory = '';
+
+	/**
+	 * Cherry_Post_Type class constructor
+	 */
+	public function __construct( $core, $args = array() ) {
+		$this->defaults = $args;
+		$this->module_directory = $core->settings['base_dir'] . '/modules/cherry-post-types';
+
+		if ( ! class_exists( 'Cherry_Post_Type' ) ) {
+			require_once( $this->module_directory . '/cherry-post-type.php' );
+		}
+	}
+
+	/**
+	 * Create new Post Type.
+	 *
+	 * @param string $slug The post type slug name.
+	 * @param string $plural The post type plural name for display.
+	 * @param string $singular The post type singular name for display.
+	 * @param array $args The custom post type arguments.
+	 * @throws Exception
+	 * @return Cherry_Post_Type
+	 */
+	public function create( $slug, $plural, $singular, $args = array() ) {
+		$params = compact('slug', 'plural', 'singular');
```

Comment:
`compact` is available since PHP 4. It should probably be replaced with an array definition to match the system requirements (PHP 5.2.4+).